options.go

```diff
@@ -642,15 +642,15 @@ var (
 )
 
 // Validate validates the fields and sets the default values.
-func (o *GrepOptions) Validate(w *Worktree) error {
+func (o *GrepOptions) Validate(r *Repository) error {
```

Comment:
Would this not break backwards compatibility?

Reply:
It's an API change, so probably yes. I overlooked this one. We could change it to `interface{}` and then check the type and return an error if type is not one of `*Repository` or `*Worktree`. AFAICT, `Validate()` is only used once in the codebase.

Reply:
I think it is easier if we simply revert this change, and then get `func (r *Repository) Grep(opts *GrepOptions)` to call `r.wt.Grep(opts)`, instead of the other way around. WDYT?

Reply:
`r.wt` is `nil` for bare repositories. We can't do that if we want to support grep on both worktree and bare repos

Reply:
To support this while not breaking the current API, we could:

```suggestion
// TODO: deprecate in favor of Validate(r *Repository) in v6.

// Validate validates the fields and sets the default values.
func (o *GrepOptions) Validate(w *Worktree) error {
	return o.validateRepository(w.r)
}

func (o *GrepOptions) validateRepository(r *Repository) error {
```